lightningnetwork/lnd
# [WIP] Show channel funding TX expiry in listchannels

Resolves #7434 

This is still a work in progress as it's not fully tested. Putting this up early for feedback. Feel free to be brutally honest. I'm new to LND and Go Lang, so still stumbling my way around.

This adds the recommended field `funding_expiry_blocks` to the `Channel` message and then computes the value based on whether the channel `IsPending`, and then uses the broadcast height of the funding transaction, the height of `GetBestBlock`, and `MaxWaitNumBlocksFundingConf`.

At the moment, this is computed in `rpcserver.createRPCOpenChannel` as this seemed to be where the `lnrpc.Channel` is created from a `channeldb.OpenChannel`. I'm a little suspicious whether it's okay to be calling `r.server.cc.ChainIO.GetBestBlock` on every channel, so if that's not right, please let me know.

Also not clear if this should be on `lncli listchannels` or on `lncli pendingchannels`. Intuitively, once the funding transaction is confirmed this field doesn't provide any information.

## Manual Testing:
- Created a simnet between alice and bob and did `lncli openchannel`.
- After generating 3 blocks on `btcd`, I could verify the `funding_expiry_blocks` using `lncli listchannels`.
- However, the channel does not show up at all in `lncli listchannels` until at least 3 blocks are generated. As such, I could not see a value other than -1 for `funding_expiry_blocks` without hacking up the logic to always compute the value.
- On the other hand, before 3 blocks are generated, the pending channel does show up using `lncli pendingchannels`, but the `funding_expiry_blocks` is not part of the response :(

Please let me know if the approach here is correct.

## Next steps:
- Write tests. I'm looking at `ht.QueryChannelByChanPoint` and either directly adding to `testListChannels` or writing an independent test.
